### ISSUE TITLE
Issue with RevertIf

### ISSUE THREAD

**Issue by EricFriedlander** [BoPeng/simuPOP#77]
Hi Bo,

I'm trying to write a simulation that restarts every time the selected allele goes extinct. However, it seems that the alleles are not reverting to their original frequencies. My code is below. Do you see what I'm doing wrong?

```
N = 1000
num_loci = 101
s = .005
totalR = .04
r = totalR/num_loci
rho = totalRho/num_loci
mrate = 10**(-8)
betas = mrate*4*N
selLocus = round((num_loci)//2)
loci = [i for i in range(num_loci)]

pop = sim.Population(size=N, loci=loci, infoFields='fitness')
for i in range(num_loci):
    init_freq = 0
    if i != selLocus:
        while init_freq < 1 / (2*N) or init_freq > 1 - (1/(2*N)):
            init_freq = np.random.beta(betas, betas, size=1)[0]
    sim.initGenotype(pop, prop = [1-init_freq, init_freq], loci = i)

sim.initSex(pop, sex=[sim.MALE, sim.FEMALE], subPops=0)
pop.dvars().freqA = []
sim.stat(pop, alleleFreq=loci)
print(pop.dvars().alleleFreq[0][1])
pop.save('init.pop')
g = pop.evolve(
            preOps=[
                sim.PointMutator(loci=selLocus, allele=1, inds = 0, ploidy=0, at=0),
            sim.Stat(alleleFreq=loci),
            sim.PyEval(r'"%d \n" % len(freqA)'),
            sim.PyEval( 'alleleFreq[0]'),   
            # sim.PyEval('alleleFreq[%d]' % selLocus),        
                            
                sim.MapSelector(loci=selLocus,
                                fitness={(0, 0): 1, (0, 1): 1+s,
                                            (1, 1): 1+2*s}),
                        sim.SNPMutator(u=mrate, v= mrate)
                    ],
            matingScheme=sim.RandomMating(ops=sim.Recombinator(rates=r)),
            postOps=[sim.Stat(alleleFreq=loci),
                        sim.PyExec('freqA.append(alleleFreq)'),
                        sim.PyEval (r'"now gen %d\n"% gen'),
                        sim.TerminateIf('gen == %d' % gens),
                        sim.RevertIf('alleleFreq[%d][1] == 0' % selLocus, "init.pop", begin=0)],
            finalOps=[sim.Stat(alleleFreq=loci), 
                    sim.PyExec('freqA.append(alleleFreq)')
                    ],
                    gen=gens)
````

**Comment by BoPeng** [comment]
`RevertIf` was some experiment that I should have deprecated earlier. The idea was sound but I found that it can be confusing to use it with other operators because of the change of `gen`, and some operators actually have some sort of "cache" that do not work with jump of gen. I am also not sure if `RevertIf` was implemented correctly because it was never thoroughly tested.

I might be able to figure out what exactly went wrong with your code or with the `RevertIf` operator but generally speaking I would suggest that you do something like this:

```
pop = ....

while True:
    pop1 = pop.clone()
    pop1.evolve(
      TerminateIf('alleleFreq[%d][1] == 0' % selLocus)
   )
```
This method would be faster because it does not need to load a population from disk.
